File: bms/go/geo/geo_s1_Angle_E7/geo_s1_Angle_E7.go
package test

import (
	"math"
)

const (
	geo_s1_Radian geo_s1_Angle = 1
	geo_s1_Degree              = (math.Pi / 180) * geo_s1_Radian
)

func (a geo_s1_Angle) E7() int32 { return geo_s1_round(a.Degrees() * 1e7) }

func (a geo_s1_Angle) Degrees() float64 { return float64(a / geo_s1_Degree) }

func geo_s1_round(val float64) int32 {
	if val < 0 {
		return int32(val - 0.5)
	}
	return int32(val + 0.5)
}

type geo_s1_Angle float64

File: translations_only/fail/go/geo/geo_s1_Angle_E7/claude2/Hinted/geo_s1_Angle_E7.rs

use std::f64::consts::PI;

const GEO_S1_RADIAN: geo_s1_angle = 1.0;
const GEO_S1_DEGREE: geo_s1_angle = (PI / 180.0) * GEO_S1_RADIAN;

fn e7(a: geo_s1_angle) -> i32 {
    geo_s1_round(degrees(a) * 1e7)  
}

fn degrees(a: geo_s1_angle) -> f64 {
    a / GEO_S1_DEGREE as f64
}

fn geo_s1_round(val: f64) -> i32 {
    if val < 0.0 {
        ((val - 0.5) as i32)
    } else {
        ((val + 0.5) as i32)
    }
}

struct GeoS1Angle(f64);

type geo_s1_angle = f64;
